[TEST] comment service, repository test 작성 + controller test 추가

Kimseungin0529 opened the pull request:
test 코드 작성
과거 dev에서 pull 해서 test 커밋만 변경사항 있음.
(+ controller test 추가)

## 개요
<!--- 변경  사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

<!---- Resolves: #(Issue Number) -->

## PR 유형
어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다. [Commit message convention](https://jane-aeiou.tistory.com/93) 참고
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

Kimseungin0529 committed:
rafactor : 댓글 생서 API BODY 수정

String -> CommentCreateResponseDto 반환

Kimseungin0529 committed:
충돌 해결

Kimseungin0529 committed:
refactor : 메서드명 변경

Kimseungin0529 committed:
Merge branch 'dev' of https://github.com/TeamCOMO/server into refactor/comment

Kimseungin0529 committed:
Merge branch 'dev' of https://github.com/TeamCOMO/server into refactor/comment

Kimseungin0529 committed:
test/refactor : CommntService test 작성 / 메소드명 수정, dto,entity 일부 수정 및 추가

1. deleteCommentWithReplies 테스트(댓글 삭제 시, 댓글에 대한 대댓글도 삭제된다) test는 에러 해결x -> 추후 재작성
2. dto, entity 수정 및 추가 -> 로직 변경은 없음

Kimseungin0529 committed:
test : CommentRepository test 작성

its-sky commented on src/main/java/project/como/domain/comment/controller/CommentController.java:
```diff
@@ -16,21 +17,21 @@
+    public ResponseEntity<CommentCreateResponseDto> createComment(@PathVariable("post_id") Long postId
             , @RequestBody @Valid CommentCreateRequestDto dto, @CurrentUser String username) { //@CurrentUser를 통해 인증된 username 가져옴
-        commentService.writeComment(username, postId, dto);
+        CommentCreateResponseDto commendDto = commentService.create(username, postId, dto);
```
commentDto라고 네이밍하는게 더 맞지 않을까?

Kimseungin0529 replied:
CommentCreateResponseDto를 commentDto로 바꾸라는거죠??

its-sky commented on src/main/java/project/como/domain/comment/repository/CommentRepository.java:
```diff
@@ -14,5 +15,5 @@ public interface CommentRepository extends JpaRepository<Comment, Long> {
     @Modifying
     @Query("DELETE FROM Comment c WHERE c.post.id = :postId")
-    void deleteAllByPostId(Long postId);
+    void deleteAllByPostId(@Param("postId") Long postId);
```
@Param 어노테이션을 붙인 이유가 있어??

Kimseungin0529 replied:
test 돌려보니까 @param for query method parameters, or when on java 8+ use the javac flag -parameters 관련 에러 로그가 나와서 @param 붙었어요

its-sky commented on src/main/java/project/como/domain/comment/service/CommentServiceImpl.java:
```diff
@@ -31,15 +32,14 @@ public class CommentServiceImpl implements CommentService {
         if(dto.getParentId() != null) {
             findParent = commentRepository.findById(dto.getParentId()).orElseThrow(() -> new CommentNotFoundException(dto.getParentId()));
-
+            System.out.println("댓글 깊이 체크");
```
이런 직접적인 print문보다 Slf4j 로깅을 하는게 더 나을듯??

Kimseungin0529 replied:
지우든가 로깅 사용으로 바꿀게요

its-sky commented on src/main/java/project/como/domain/user/model/User.java:
```diff
@@ -41,7 +42,8 @@ public class User implements UserDetails {
 		return this.roles.stream()
-				.map(SimpleGrantedAuthority::new).toList();
+				.map(SimpleGrantedAuthority::new)
+				.collect(Collectors.toList());
```
여기서 toList()로 하게 되면 수정불가능한 List로 만들게 되고 .collect(Collectors.toList())로 하게 되면 수정 가능한 리스트로 만들어지게 되는데 이후에 리스트를 조작하는 로직이 없으면 간결하게 toList()로 해도 될거 같아

Kimseungin0529 replied:
넵넵

its-sky commented on src/test/java/project/como/domain/comment/repository/CommentRepositoryTest.java:
```diff
@@ -0,0 +1,125 @@
+    }
+    @AfterEach
+    void close(){
+        postRepository.delete(post1); // 만약 운영 DB를 사용한다면 deletAll을 하면 안됨?
```
내가 알기로는 delete랑 deleteAll의 차이는 그냥 없다고 보면 돼. deleteAll은 for문 돌면서 하나씩 삭제하는 걸로 알고 있어. 그 대체재로 벌크 삭제를 할 떄 deleteAllInBatch라는 메소드가 있어 그거로 하면 될거야.

its-sky commented on src/test/java/project/como/domain/comment/service/CommentServiceTest.java:
```diff
@@ -0,0 +1,377 @@
+                .isInstanceOf(CommentLevelExceedException.class);
+                //.hasMessage("대댓글 깊이는 최대 3입니다. 해당 댓글에 대댓글을 작성할 수 없습니다.");
+        /**
+         * 이 친구도 message 값이 null이라고 나옴...
```
이런 것들은 다시 코드 봐서 고치는게 좋을거 같아!

Kimseungin0529 replied:
확인해보니 ComoException 클래스에서 상속받은 RuntimeException에 super(detail) 값을 넘겨주지 않아서 message 값 확일할 때, null로 나오더라고요
그래서 public ComoException(){
     super(this.detail);
}로 넘기려 했는데 기존 생성자랑 충돌나요!
예외 처리 핸들링 생각하면 기존 생성자도 필요한데 혼자 막 바꾸기는 그래서 일단 넘어가겠슴돠

its-sky reviewed (commented):
전체적으로 잘했고 JUnit5 같은 통합테스트도 좋은데 이런 통합테스트들은 의존성 설정같은 것들이 코드가 커지면서 시간이 오래 걸려. 그래서 Mockito 같은 단위 테스트 코드도 공부하는게 좋을거야~ 수고했어

Kimseungin0529 committed:
병합 충돌 commit

Kimseungin0529 committed:
Merge branch 'dev' into refactor/comment

Kimseungin0529 committed:
Merge branch 'refactor/comment' of https://github.com/TeamCOMO/server into refactor/comment

Kimseungin0529 committed:
refactor : 리뷰 반영

Kimseungin0529 committed:
refactor : 리뷰 반영

Kimseungin0529 committed:
refactor/test : comment 대댓글 관련 연관관계 메서드 사용 -> test 오류 해결/댓글 생성 API test 추가

대댓글 생성 시, 부모 댓글에 대댓글을 추가해줘야하는데 해당 기능이 없었다. postman 사용 시, 의도대로 삭제는 됐지만 해당 기능이 없어 실패.   '댓글 삭제 시, 댓글에 대한 대댓글도 삭제된다.'
test 성공 -> 해당 기능 추가, orphanRemoval = true로 (자식)대댓글도 삭제

Kimseungin0529 committed:
Merge branch 'refactor/comment' of https://github.com/TeamCOMO/server into refactor/comment

Kimseungin0529 committed:
refactor : 댓글 생성 API test 수정

Kimseungin0529 committed:
test : 댓글 생성, 목록 조회, 수정, 삭제 API test 추가

test -> 성공 케이스와 dto validation으로 인한 실패 케이스

Kimseungin0529 committed:
Merge branch 'dev' of https://github.com/TeamCOMO/server into refactor/comment
